h/api/nipsa/worker.py

```diff
@@ -0,0 +1,69 @@
+"""Worker functions for the NIPSA feature."""
+import json
+
+import annotator
+
+from h.api import search
+from h.api.nipsa import search as nipsa_search
+
+
+def add_nipsa_action(annotation):
+    """Return an Elasticsearch action for adding NIPSA to the annotation."""
+    return {
+        "_op_type": "update",
+        "_index": annotator.es.index,
+        "_type": "annotation",
+        "_id": annotation["_id"],
+        "doc": {"not_in_public_site_areas": True}
+    }
+
+
+def remove_nipsa_action(annotation):
+    """Return an Elasticsearch action to remove NIPSA from the annotation."""
+    return {
+        "_op_type": "update",
+        "_index": annotator.es.index,
+        "_type": "annotation",
+        "_id": annotation["_id"],
+        "script": "ctx._source.remove(\"not_in_public_site_areas\")"
+    }
+
+
+def add_or_remove_nipsa(userid, action, es_client):
+    """Add/remove the NIPSA flag to/from all of the user's annotations."""
+    assert action in ("nipsa", "unnipsa")
+
+    if action == "nipsa":
+        query = nipsa_search.not_nipsad_annotations(userid)
+    else:
+        query = nipsa_search.nipsad_annotations(userid)
+
+    annotations = search.scan(es_client=es_client, query=query, fields=[])
+
+    if action == "nipsa":
+        actions = [add_nipsa_action(a) for a in annotations]
```

Comment:
Could set a local variable above to the value `add_nipsa_action` or `remove_nipsa_action` and save the duplicated conditional. Might read more cleraly.